### DISCUSSION TITLE
[Support] 怎么获取所有的controllers

### DISCUSSION THREAD

**Discussion by zhanzhan0720** [midwayjs/midway#4070] [Q&A]
### Describe the problem（描述问题）
想获取所有的controllers， 这样获取不到呢

async onReady(container: IMidwayContainer) {
    const controllers = await container.getAsync('controllers');
    console.log('controllers', controllers);
}

### Midway Versions（Midway 版本）

Need to install the following packages:
midway-version@1.4.0

**Comment by waitingsong** [comment]
如果是获取路由，参考 https://midwayjs.org/docs/built_in_service#midwaywebrouterservice

**Comment by czy88840616** [comment]
通过 listModule

```ts

import { CONTROLLER_KEY, listModule } from '@midwayjs/core';

const modules = listModule(CONTROLLER_KEY);

```
